var/spack/repos/builtin/packages/googletest/package.py

```diff
@@ -50,6 +50,7 @@ def cmake_args(self):
             # New style (contains both Google Mock and Google Test)
             args.append(self.define("BUILD_GTEST", True))
             args.append(self.define_from_variant("BUILD_GMOCK", "gmock"))
+            args.append(self.define("CMAKE_POSITION_INDEPENDENT_CODE", True))
```

Comment:
This should only matter if `~shared` because shared libraries always use PIE/PIC. Can you provide more details on the use case with static libraries that require PIC?

Reply:
Hi @sethrj ,
rocm-validation-suite@5.3.0 spec (
[rocm-validation-suite-5.3.0-spec.txt](https://github.com/spack/spack/files/10317044/rocm-validation-suite-5.3.0-spec.txt)
) includes ^googletest@1.12.1%gcc@8.3.1+gmock~ipo+pthreads~shared build_system=cmake build_type=RelWithDebInfo cxxstd=11 arch=linux-centos8-skylake_avx512

Reply:
@renjithravindrankannath Do you know why it explicitly chooses static but happens to get built into another shared library (which is the only situation where the PIC flag should matter)? And again, this change should only apply when `~shared`.